hkmc2/shared/src/test/mlscript/bbml/bbDisjoint.mls

```diff
@@ -113,29 +113,76 @@ region x in
 //│ Type: Pair[out Ref[Int, ?], out Ref[Int, ?]]
 
 
-fun anno: Int -> @outer Int
+region x in
+  (region y in let t = helper(y) in 42): [A] -> Int
+//│ Type: Int
+
+
+region x in
+  region y in
+    let t = helper(x) in 42
+//│ Type: Int
+
+
+:e
+region x in
+  (region y in let t = helper(x) in 42): [A] -> Int
+//│ ╔══[ERROR] Type error in reference with expected type 'r1
+//│ ║  l.129: 	  (region y in let t = helper(x) in 42): [A] -> Int
+//│ ║         	                              ^
+//│ ╟── because: cannot constrain  Region[x]  <:  'r1
+//│ ╟── because: cannot constrain  Region[in x out x]  <:  'r1
+//│ ╟── because: cannot constrain  Region[in x out x]  <:  Region[in 'reg out 'reg1]
+//│ ╟── because: cannot constrain  x  <:  'reg1
+//│ ╟── because: cannot constrain  x  <:  'reg1
+//│ ╟── because: cannot constrain  x  <:  'env
+//│ ╟── because: cannot constrain  x  <:  'env
+//│ ╙── because: cannot constrain  x  <:  env@outer ∨ y
+//│ ╔══[ERROR] Type error in region expression with expected type forall 'A: Int
+//│ ║  l.129: 	  (region y in let t = helper(x) in 42): [A] -> Int
+//│ ║         	                       ^^^^^^^^^^^^^^^
+//│ ╟── because: cannot constrain  'eff  <:  ⊥
+//│ ╟── because: cannot constrain  'eff  <:  ¬()
+//│ ╟── because: cannot constrain  (¬⊥ ∧ ¬'y1) ∧ x  <:  ¬()
+//│ ╟── because: cannot constrain  x  <:  'y1
+//│ ╙── because: cannot constrain  x  <:  ¬()
+//│ Type: Int
+
+
+fun anno: [outer A] -> Int ->{A} Int
+//│ Type: ⊤
+
+
+fun anno2: [outer] -> Int ->{outer} Int
 //│ Type: ⊤
 
 
 :e
-42: @outer Int
-//│ ╔══[ERROR] Illegal outer binding.
-//│ ║  l.121: 	42: @outer Int
-//│ ╙──       	     ^^^^^^^^^
+fun badanno: outer
+//│ ═══[ERROR] Illegal outer reference.
+//│ ═══[ERROR] Invalid type
+//│ Type: ⊤
+
+:e
+fun badanno2: [outer A, outer B] -> Int ->{A | B} Int
+//│ ╔══[ERROR] Only one outer variable can be bound.
+//│ ║  l.167: 	fun badanno2: [outer A, outer B] -> Int ->{A | B} Int
+//│ ╙──       	              ^^^^^^^^^^^^^^^^^^
+//│ ╔══[ERROR] Illegal forall annotation.
+//│ ║  l.167: 	fun badanno2: [outer A, outer B] -> Int ->{A | B} Int
+//│ ╙──       	              ^^^^^^^^^^^^^^^^^^
 //│ ═══[ERROR] Invalid type
-//│ ╔══[ERROR] Type error in integer literal with expected type ⊥
-//│ ║  l.121: 	42: @outer Int
-//│ ║         	^^
-//│ ╙── because: cannot constrain  Int  <:  ⊥
-//│ Type: ⊥
+//│ Type: ⊤
 
 
-fun annohelper: Region[outer] ->{outer} @outer Pair[out Ref[Int, out outer], out Ref[Int, out Any]]
+fun annohelper: [outer, T extends outer] -> Region[T] ->{T} Pair[out Ref[Int, out T], out Ref[Int, out Any]]
```

Comment:
Why do we have `Any` in `Ref[Int, out Any]`?

Reply:
Because the second reference returned by `fork` is located in `r2`. It is extruded after we leave the scope of `r2`. But it can be written as `out ~outer`.

Reply:
Yes, `~outer` is what I meant in my other comment.